### PR TITLE
feat: implement Kafka sourcer with schema registry

### PR DESCRIPTION
**Pull request by KeranYang**
A working sourcer with unit tests added. Tested by reading from my test topic `numagen` from my sink implementation and verified that messages are properly deserialized using schema registry and show up in the log sink.

I have a bunch of TODOs and FIXMEs that I will work on in follow-up PRs.

Note: a lot of the code are borrowed from [eventbus-numaflow](https://github.intuit.com/oss-analytics/eventbus-numaflow). Thank you @nagarajatantry for setting a good example. :)

### CODE REVIEW

File: src/main/java/io/numaproj/kafka/KafkaApplication.java [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
 
 @Slf4j
 @SpringBootApplication
+@EnableConfigurationProperties
+@ComponentScan(
```
Comment (nagarajatantry) [inline src/main/java/io/numaproj/kafka/KafkaApplication.java:19]: Might be better off to create a separate folder for all the source and producer related code.   Then create a separate configuration file where you can add the package you want to scan depending on source or sink.

File: src/main/java/io/numaproj/kafka/config/ProducerConfig.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  // Kafka producer client
+  @Bean
+  public KafkaProducer<String, GenericRecord> kafkaProducer() throws IOException {
+    log.info(
+        "Instantiating the Kafka producer from the producer properties file path: {}",
+        this.producerPropertiesFilePath);
+    Properties props = new Properties();
+    InputStream is = new FileInputStream(this.producerPropertiesFilePath);
```
Comment (nagarajatantry) [inline src/main/java/io/numaproj/kafka/config/ProducerConfig.java:50]: Close stream?

File: README.md [hunk 3]
```diff
@@ -9,20 +9,125 @@ Apache Kafka. It allows you to read/write data from/to Kafka using Numaflow.
...
+    # Schema Registry connection configurations
+    schema.registry.url=localhost:8081
+    basic.auth.credentials.source=USER_INFO
+    basic.auth.user.info=a:b
+    # Other configurations
+    retries=0
+    # group.id is required for consumer clients
+    group.id=group1
+  user.configuration: |
+    topicName: numagen
```
Comment (nagarajatantry) [inline README.md:52]: Do we want to support topic from different brokers? For example.. reading from 2 different regions then consumer properties and topic name will have to go hand in hand.

**Review by nagarajatantry — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (KeranYang) [reply README.md:52]: https://github.com/numaproj-contrib/kafka-java/issues/7

**Review by nagarajatantry — APPROVED** [review]
Comments will be addressed in the follow up PRs.. approving this..

**Pull request merged**